backend-lib/src/main/rust/lib.rs

```diff
@@ -198,28 +196,24 @@ pub unsafe extern "C" fn Java_cash_z_ecc_android_sdk_internal_jni_RustBackend_in
             Err(e) => Err(format_err!("Error while initializing data DB: {}", e)),
         }
     });
-    unwrap_exc_or(&env, res, -1)
+    unwrap_exc_or(&mut env, res, -1)
 }
 
-fn encode_usk(
-    env: &JNIEnv<'_>,
+fn encode_usk<'a>(
+    env: &mut JNIEnv<'a>,
     account: AccountId,
     usk: UnifiedSpendingKey,
-) -> Result<jobject, failure::Error> {
+) -> jni::errors::Result<JObject<'a>> {
     let encoded = SecretVec::new(usk.to_bytes(Era::Orchard));
     let bytes = env.byte_array_from_slice(encoded.expose_secret())?;
-    let output = env.new_object(
+    env.new_object(
         "cash/z/ecc/android/sdk/internal/model/JniUnifiedSpendingKey",
         "(I[B)V",
-        &[
-            JValue::Int(u32::from(account) as i32),
-            JValue::Object(unsafe { JObject::from_raw(bytes) }),
-        ],
-    )?;
-    Ok(output.into_raw())
+        &[JValue::Int(u32::from(account) as i32), (&bytes).into()],
+    )
 }
 
-fn decode_usk(env: &JNIEnv<'_>, usk: jbyteArray) -> Result<UnifiedSpendingKey, failure::Error> {
+fn decode_usk(env: &JNIEnv, usk: JByteArray) -> Result<UnifiedSpendingKey, failure::Error> {
```

Comment:
It's kind-of an implementation detail that this doesn't need to call `JNIEnv` methods that take `&mut self`. But since it's not public, that doesn't matter I guess.